src/fairseq2/recipes/hg/evaluator.py

```diff
@@ -172,6 +173,11 @@ def _do_run(self) -> None:
                         predictions=predictions, references=references
                     )
 
+                    del inputs
+                    del targets
+                    del outputs
+                    torch.cuda.empty_cache()
```

Comment:
What is the reason for this explicit `empty_cache()` call?

Reply:
I noticed that the data from previous batches were not freed which lead to the evaluator crashing OOM after a some batches.

Quoting https://github.com/facebookresearch/fairseq2/pull/740#issuecomment-2278592218
> I noticed that running the evaluation with `max_samples > 20` on a 15GB VRAM setup was causing a CUDA out-of-memory error. To address this, I made improvements to better manage and free up memory. With these updates, you can now run the evaluation on the entire dataset, provided the `batch_size` fits within the available memory. I successfully tested this on the whole test split (approximately 2939 examples) using `whisper-tiny` and `wav2vec2_asr_base_10h` with `max_num_elements=10`.
>...
> 
> Edit: `max_num_elements` can also be increased significantly by defining `max_audio_len` as some audio samples were quite long and the whole batch gets padded to the longest sequence causing some batches to require a lot more memory than expected.